packages/moonstone/VirtualList/VirtualListBase.js

```diff
@@ -619,11 +620,29 @@ const VirtualListBaseFactory = (type) => {
 		}
 
 		onKeyDown = (ev) => {
-			if (getDirection(ev.keyCode)) {
-				ev.preventDefault();
-				ev.stopPropagation();
+			const {keyCode} = ev;
+			const direction = getDirection(keyCode);
+
+			if (getDirection(keyCode)) {
 				Spotlight.setPointerMode(false);
-				SpotlightAccelerator.processKey(ev, this.onAcceleratedKeyDown);
+
+				if (SpotlightAccelerator.processKey(ev, nop)) {
+					ev.preventDefault();
+					ev.stopPropagation();
+				} else {
+					const {repeat, target} = ev;
+					const index = getNumberValue(target.dataset.index);
+					const {isWrapped, nextIndex} = this.getNextIndex({index, keyCode, repeat});
+
+					if (nextIndex >= 0) {
+						ev.preventDefault();
+						ev.stopPropagation();
+						this.onAcceleratedKeyDown({index, isWrapped, keyCode, nextIndex, repeat, target});
+					} else if (repeat || Spotlight.move(direction) && this.uiRefCurrent.containerRef.current.contains(Spotlight.getCurrent())) {
```

Comment:
When we reach this point - at the `else if...` statement - the value of `nextIndex` is `-1`. This means we'll need to hand off to spotlight in order to properly set focus (via the following `Spotlight.move()`). This happens because either
a) we're at the bounds of the list
b) the next item is disabled (in grid scenarios we can't always assume the next enabled item is physically closer or that it's the next eligible target according to the spotlight algorithm)

We need to consume `repeat` events when we're at the bounds of the list, since we want to stop 5way progression at the container during a hold. However, it looks like I needed some additional logic to specify that case vs when the next item is simply disabled.